src/ceph_nfs.py

```diff
@@ -0,0 +1,329 @@
+#!/usr/bin/env python3
+
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Handle Charm's NFS Client Events."""
+
+import json
+import logging
+from typing import Callable
+
+from ops.charm import CharmBase, RelationEvent
+from ops.framework import EventBase, EventSource, Object, ObjectEvents
+from ops_sunbeam.relation_handlers import RelationHandler
+
+import ceph
+import microceph
+import utils
+from ceph_broker import get_named_key
+from microceph_client import Client
+
+logger = logging.getLogger(__name__)
+
+
+class CephNfsConnectedEvent(RelationEvent):
+    """ceph-nfs connected event."""
+
+    pass
+
+
+class CephNfsDepartedEvent(RelationEvent):
+    """ceph-nfs relation has departed event."""
+
+    pass
+
+
+class CephNfsReconcileEvent(RelationEvent):
+    """ceph-nfs relation reconciliation event."""
+
+    pass
+
+
+class CephNfsEvents(ObjectEvents):
+    """Events class for `on`."""
+
+    ceph_nfs_connected = EventSource(CephNfsConnectedEvent)
+    ceph_nfs_departed = EventSource(CephNfsDepartedEvent)
+    ceph_nfs_reconcile = EventSource(CephNfsReconcileEvent)
+
+
+class CephNfsClientProvides(Object):
```

Comment:
I would love to move CephNFSClientProvides and https://review.opendev.org/c/openstack/sunbeam-charms/+/956370/4/charms/manila-cephfs-k8s/lib/charms/manila_cephfs_k8s/v0/cephfs.py as a library published by charm microceph so that ANY cephnfs clients can use the library.

@UtkarshBhatthere Any concerns here?

Reply:
So if the ceph-nfs client library is not generic enough (to be used by another client) it does not make sense to keep it in the microceph charm repository and would be best organized in the coupled client itself.

Reply:
@claudiubelu please go ahead with merging this in charm-microceph. Also please use the name `ceph-nfs-client` for the lib